src/Symfony/Component/Form/Form.php

```diff
@@ -1044,11 +1044,29 @@ public function createView(FormView $parent = null)
             $view->children[$name] = $child->createView($view);
         }
 
+        $this->sort($view->children);
+
         $type->finishView($view, $this, $options);
 
         return $view;
     }
 
+    /**
+     * Sorts view fields based on their priority value.
+     */
+    private function sort(array &$children): void
+    {
+        $c = [];
+        $i = 0;
+        foreach ($children as $name => $child) {
+            $c[$name] = ['p' => $child->vars['priority'], 'i' => $i++];
```

Comment:
should we use `?? 0` for safety in case someone adds a child which does not have such variable ? Technically, there is nothing forcing to have a type extending BaseType as the root of the hierarchy (even though doing things like that makes it much harder to use it as you will need to define lots of rendering for it if it does not have the `form` type as its last rendering fallback).
AFAICT, the Form class does not make any assumption about which variables are available in children for now. Handling that case would keep it like that.